### PR TITLE
refactor: endpoints fetching and display

### PR DESCRIPTION
**Pull request by thxCode**
**Problem:**
We have struggled with endpoint for a long time: https://github.com/seal-io/walrus/issues?q=is%3Aissue+access+endpoints+is%3Aclosed, let's figure out this today.

Walrus relies on the [Operator's GetEndpiont](https://github.com/seal-io/walrus/blob/937ca0fd19efd0e7137164f5b81c511f160426b2/pkg/operator/types/types.go#L51-L55) to discover endpoints from a resource, but the discovered endpoints fail to convey their correct intention.

**Solution:**
For Walrus, a resource endpoint should be functional and accessible. To describe the purpose and access way of an endpoint, a good choice is URL, e.g. `https://host/path/subpath`, or `s3://host/bucket`.

The endpoint is an attribute of a resource, so this PR removes the endpoint record from the resource component's status and records to the resource itself directly.

Since the active endpoint discovery cannot convey the correct intention, so this PR removes all implementations for `GetEndpoints`, and respects the returning result(output) from the resource.

Walrus built-in ContainerService collection supports describing the endpoint's purpose with the schema on ports.
- https://github.com/walrus-catalog/terraform-kubernetes-containerservice/commit/8893c34c0ff2a8e06341f21f154bf54e0fceb2f9
- https://github.com/walrus-catalog/terraform-docker-containerservice/commit/6258147d2eec227175796ed71ab968e3432c6e43

This PR parses the endpoint from an output named `walrus_endpoints` or `endpoints`, and only accepts the `map(string)` type.
- `walrus_endpoints` gives the capability to be compatible with original `endpoints`.

**BREAKING CHANGES**
- route change from `/v1/projects/{project}/environments/{environment}/resources/{resource}/access-endpoints` to `/v1/projects/{project}/environments/{environment}/resources/{resource}/endpoints`, and response change `[{"name":"","endpoints":["",""]}]` to `[{"name":"", "url":""}]`.

**CHORES**
- function style: receiver, comment, and purpose. 

**Related Issue:**
#1755

### CODE REVIEW

File: pkg/apis/resource/extension.go [hunk 1]
```diff
@@ -224,9 +217,9 @@ func (h Handler) RouteStop(req RouteStopRequest) error {
 	return pkgresource.Stop(req.Context, req.Client, entity, opts)
 }
 
-func (h Handler) RouteGetAccessEndpoints(req RouteGetAccessEndpointsRequest) (RouteGetAccessEndpointsResponse, error) {
+func (h Handler) RouteGetEndpoints(req RouteGetEndpointsRequest) (RouteGetEndpointsResponse, error) {
```
[the remark this answers is not in the document]
Comment (thxCode) [reply pkg/apis/resource/extension.go:220]: this is no single resource watch api.

File: pkg/deployer/terraform/parse.go [hunk 2]
```diff
@@ -354,20 +354,27 @@ func getServiceDependencyOutputs(
...
 		}
 
-		for n, o := range revisionOutput {
+		for n, o := range osm {
 			if _, ok := dependOutputs[n]; !ok {
 				continue
 			}
 
-			outputs[n] = o
+			// FIXME(thxCode): mitigate parser.OutputState to types.OutputValue.
```
Comment (gitlawr) [inline pkg/deployer/terraform/parse.go:372]: Mitigate or migrate? Is it hard to include this refactoring in this pr?

Comment (thxCode) [reply pkg/deployer/terraform/parse.go]: updated

**Review by alexcodelf — APPROVED** [review]
LGTM

**Review by gitlawr — APPROVED** [review]

**Pull request merged**